File: android/app/src/main/java/com/example/yugiohscanner/cloud/CopyLocation.kt
package com.example.yugiohscanner.cloud

/**
 * Spec B1 §7.3/§10 (Task 10): Standort-Chip-Text fuer ein Exemplar -- ZEICHENGLEICH zur
 * Desktop-Fassung (`desktop/src/utils/copyLocation.js#formatCopyLocation`), damit ein Standort
 * auf beiden Geraeten gleich aussieht. Wer hier etwas aendert, aendert dort mit.
 *
 * Rein und ohne Compose-Abhaengigkeit -- eigene Datei statt in CardDetailScreen/CollectionScreen
 * verschachtelt, damit sie ohne Robolectric als JVM-Unit-Test lauffaehig bleibt (siehe
 * CopyLocationTest), genau wie Tags/ScanAggregator/RarityRank es in diesem Projekt vormachen.
 *
 * `container` ist die zum Exemplar gehoerende Zeile aus ContainersRepository.list() (oder null,
 * wenn keine gefunden wird -- z.B. waehrend Behaelter noch nachgeladen werden).
 */
object CopyLocation {
    fun format(copy: CopyRow, container: ContainerRow?): String {
        if (copy.containerId == null || container == null) return "—"
        if (copy.page != null && copy.slot != null) return "${container.name} · S${copy.page} · F${copy.slot}"
        return container.name
    }
}

File: desktop/src/utils/copyLocation.js
// Spec B1 §7.3: Standort-Chip-Text fuer ein Exemplar -- eine Funktion fuer alle Darstellungsorte
// (CardDetailPanel.jsx, die Sammlungsliste aus Task 7 importiert sie von hier, das Handy aus
// Task 10 baut sie zeichengleich nach), sonst weicht die Formatierung irgendwann auseinander.
// `container` ist die zum Exemplar gehoerende Zeile aus listContainers() (oder undefined/null,
// wenn keine gefunden wird -- z.B. waehrend Behaelter noch nachgeladen werden).
export function formatCopyLocation(copy, container) {
  if (!copy?.container_id || !container) return '—';
  if (copy.page != null && copy.slot != null) return `${container.name} · S${copy.page} · F${copy.slot}`;
  return container.name;
}
